openqa-advanced-retrigger-jobs

```diff
@@ -1,16 +1,105 @@
-    $client_prefix openqa-cli api --host "$host" -X POST jobs/"$i"/restart
-    [ -n "$comment" ] && $client_prefix openqa-cli api --host "$host" -X POST jobs/"$i"/comments text="$comment"
-done
+#!/usr/bin/env python3
+
+"""
+Retrigger openQA jobs based on database queries.
+
+Needs SSH access to the specified target openQA host.
+
+Simple example call retriggering all recent incompletes on the default host:
+    %(prog)s
+
+Advanced example retriggering failed instead of incompletes, verbose output, with custom starting date,
+custom host and excluding jobs with \":investigate:\" in the name, executed as dry-run:
+    %(prog)s -vvvv --host openqa.example.org --failed-since '2000-01-01T10:00' --result failed \
+--additional-filters \"test not like '%%:investigate:%%'\" --dry-run
+"""
+
+import argparse
+import logging
+import subprocess
+import sys
+from datetime import datetime
+
+logging.basicConfig()
+log = logging.getLogger(sys.argv[0] if __name__ == "__main__" else __name__)
+
+
+class CustomFormatter(argparse.ArgumentDefaultsHelpFormatter, argparse.RawDescriptionHelpFormatter):
+    """Preserve multi-line __doc__ and provide default arguments in help strings."""
+
+    pass
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description=__doc__, formatter_class=CustomFormatter)
+    parser.add_argument(
+        "-v",
+        "--verbose",
+        help="Increase verbosity level, specify multiple times to increase verbosity",
+        action="count",
+        default=1,
+    )
+    parser.add_argument("-H", "--host", default="openqa.opensuse.org", help="Target openQA host")
+    parser.add_argument(
+        "-s",
+        "--failed-since",
+        default=datetime.today().isoformat(),
+        help="Filter jobs failed since this date",
+    )
+    parser.add_argument("-w", "--worker", default=None, help="Filter jobs assigned to this worker")
+    parser.add_argument("-i", "--instance", default=None, help="Instance of the worker")
+    parser.add_argument("-r", "--result", default="incomplete", help="Filter jobs with this result")
+    parser.add_argument(
+        "-a",
+        "--additional-filters",
+        default=None,
+        help="Additional filters for the SQL query",
+    )
+    parser.add_argument("-c", "--comment", default=None, help="Comment to add to the retriggered jobs")
+    parser.add_argument(
+        "-d",
+        "--dry-run",
+        action="store_true",
+        help="If set, only print the actions without executing",
+    )
+    args = parser.parse_args()
+    verbose_to_log = {
+        0: logging.CRITICAL,
+        1: logging.ERROR,
+        2: logging.WARN,
+        3: logging.INFO,
+        4: logging.DEBUG,
+    }
+    logging_level = logging.DEBUG if args.verbose > 4 else verbose_to_log[args.verbose]
```

Comment:
Maybe this is not very explicit but I like it (considering numerical values for log levels are [documented explicitly](https://docs.python.org/3/library/logging.html#logging-levels)):
```suggestion
    logging_level = (5 - min(args.verbose, 4)) * 10
```